### PR TITLE
Add Enhancement: Fix undo and redo view behaviour

### LINKED ISSUE #232: [PE-D][Tester C] Undo command always returns back to list all members and events
Now, the only enrolment is member 1 - event 1.
After unenrolling m1-e1:

After undoing:

This actually lists all members and events, but as a user, I may be confused to think that member 2 is enrolled to event 1, which is not what I expect.

I suggest `undo` and `redo` command can apply the same filtering conditions on the displayed list as before, rather than always listing everything. In that way, it's also more clear what has been undone/redone especially for undoing/redoing `enrol` or `unenrol` commands.

-------------
Labels: `type.FeatureFlaw` `severity.High`
original: Singa-pirate/ped#7

### PR DESCRIPTION
**Pull request by dylkaw**
Closes #232 

What have you done in this PR?
* Add planned enhancement for undo and redo commands to remain at the current view upon execution

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -450,6 +450,18 @@ We propose to make the index error messages more specific and highlight to the u
...
 the lines of "The provided Event Index is not a non-zero unsigned integer."
 
+### Make UI stay on current view upon undo/redo
+**Current Implementation:**
+* **Current Issue:** Currently, the `undo` and `redo` commands will always result in all members and events being listed.
+* **Example:** For example, if the user enrols a member to an event and views the member, before deciding to unenrol the member
+from the event by executing the `undo` command, all the members and events would then be listed. This inconveniences the user
+as they will then have to execute the `viewMember` command again to see their changes. 
+
+**Proposed solution:**
```
Comment (wr1159) [inline docs/DeveloperGuide.md]: I believe there is currently an PR to change `Proposed solution` to `Proposed Solution`, should we standardise it here now?

**Review by wr1159 — APPROVED** [review]
LGTM just a small comment about the casing of `Proposed Solution`

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (SelwynAng) [reply docs/DeveloperGuide.md]: Yes agreed.

Comment (dylkaw) [reply docs/DeveloperGuide.md]: Done!

**Pull request merged**